pkg/apis/config/types.go

```diff
@@ -87,6 +87,12 @@ type TargetLoadPackingArgs struct {
 	DefaultRequestsMultiplier string
 	// Node target CPU Utilization for bin packing
 	TargetUtilization int64
+	// Types of metric provider to be used: KubernetesMetricsServer, Prometheus, SignalFx
+	MetricProviderType string
```

Comment:
`MetricsProviderType`, `MetricsProviderAddress` and `MetricsProviderToken`. here and elsewhere for consistency.

Reply:
Let's all use `MetricProvider` as we should not use plural form when `metric` is used as an attributive.

Reply:
I see, not a grammar expert but I only used `Metrics` because it is used so elsewhere [[1](https://github.com/DataDog/datadog-agent/blob/c4f38af1897bac294d8fed6285098b14aafa6178/docs/cluster-agent/CUSTOM_METRICS_SERVER.md#datadog-cluster-agent--custom--external-metrics-provider)] [[2](https://kubernetes.io/docs/tasks/debug-application-cluster/resource-metrics-pipeline/#metrics-server)] [[3](https://github.com/zalando-incubator/kube-metrics-adapter)]. If you would like to use `Metric` please modify it in KEP and load-watcher repo/code appropriately.